src/tlo/methods/stunting.py

```diff
@@ -484,7 +484,8 @@ def apply_model(self, model, mask, days_exposed_to_risk):
         df = self.sim.population.props
         rng = self.module.rng
 
-        annual_prob = model.predict(df.loc[mask]).clip(upper=1.0)
+        annual_prob = model.predict(df.loc[mask]).clip(upper=0.99999)  # Prevents the np.log call below raising warning
+        #                                                                if it encounters np.log(0.0).
         cum_prob_over_days_exposed = 1.0 - np.exp(np.log(1.0 - annual_prob) * days_exposed_to_risk / DAYS_IN_YEAR)
```

Comment:
Is there a reason for favouring the form here over the more direct and I think equivalent

```suggestion
        cum_prob_over_days_exposed = 1.0 - (1.0 - annual_prob) ** (days_exposed_to_risk / DAYS_IN_YEAR)
```
(as $\exp(\log(1-p) \times r) \equiv (1-p)^r$).

Switching to the above form I think would eliminate the need to clip at 0.99999 as we would not be computing a logarithm, and I think should be relatively numerically stable given `annual_prob` is by definition in range `[0, 1]`.

Reply:
Good point. I think it is actually more correct.